docs/tutorials/achievement_icons.md

```diff
@@ -2,68 +2,76 @@
 
 This quick tutorial will cover to how get achievement icons from Steam's servers. It was made because there are some extra steps folks need to use to render the images and it may not be very clear otherwise.
 
+The relevant Steam classes and functions for this tutorial are:
+
+  * [User Stats class](../classes/user_stats.md)
+    * [requestCurrentStats()](../classes/user_stats.md#requestcurrentstats)
+    * [getAchievementIcon()](../classes/user_stats.md#getachievementicon)
+  * [Utils class](../classes/utils.md)
+    * [getImageSize()](../classes/utils.md#getimagesize)
+	* [getImageRGBA()](../classes/utils.md#getimagergba)
+
 ---
 
-## Getting The Handle And Buffer
+## Prerequisites
+
+Before you can get achievement icons you first need to have retrieved the user's stats from Steam. This is done by default when initializing Steam, but can optionally be disabled. This tutorial assumes that you have already received the user stats. See [requestCurrentStats()](../classes/user_stats.md#requestcurrentstats) for more information if needed.
 
-First you will need to request the icon handle by using the achievement's name you set in the Steamworks back-end:
+## Getting the Handle and Buffer
+
+First you will need to request the icon handle by using the achievement's API name you set in the Steamworks back-end:
 
 ````
-var HANDLE: int = Steam.getAchievementIcon("achievement name")
+var HANDLE: int = Steam.getAchievementIcon("ACHIEVEMENT_NAME")
 ````
 
-This handle is used to get the image information. This would be passed to a buffer variable like so:
+This handle is an ID used to get information about the image's size and a byte array (buffer) of the pixel data:
 
 ````
+var SIZE: Dictionary = Steam.getImageSize(HANDLE)
 var BUFFER: Dictionary = Steam.getImageRGBA(HANDLE)
 ````
 
+Note that the `getImageRGBA()` function can be somewhat expensive so it is recommended to only call this once per image handle and cache the result if you will need to access the image data multiple times.
+
 ---
 
-## Creating The Image
+## Creating the Image
 
-This buffer contains the actual image data for our icon. However, as this buffer is simply binary data, we need to load it into an Image so that Godot can use it as a Texture:
+This buffer contains the actual image data for our icon. However, as it is simply binary data, we need to load it into an Image so that Godot can use it as a Texture. The format of the data in the buffer is RGBA8 when we receive it from Steam, so we need to tell Godot how to make sense of it by specifying it as such:
 
 === "Godot 2.x, 3.x"
 	````
-	var SIZE: int = 64
 	var ICON: Image = Image.new()
-	var ICON_TEXTURE: ImageTexture = ImageTexture.new()
-
+	ICON.create_from_data(SIZE.width, SIZE.height, false, Image.FORMAT_RGBA8, BUFFER)
 	````
 === "Godot 4.x"
 	````
-	var SIZE: int = 64
+	var ICON := Image.create_from_data(SIZE.width, SIZE.height, false, Image.FORMAT_RGBA8, BUFFER)
```

Comment:
I almost never use `:=` for anything as I try to keep everything super basic for all users.  Also trying to keep everything consistent; either everything is short-hand or everything is explicit.

Reply:
That line of code in particular was too long to display without horizontal scrolling, so I tried to shorten it with the `:=`. It's still too long even with that, so I guess the change was pointless in the end. I'll change it back.